Python/import.c

```diff
@@ -822,7 +959,8 @@ exec_code_in_module(PyObject *name, PyObject *module_dict, PyObject *code_object
     }
     Py_DECREF(v);
 
-    if ((m = PyDict_GetItem(modules, name)) == NULL) {
+    m = _PyImport_GetModule(name);
+    if (m == NULL) {
         PyErr_Format(PyExc_ImportError,
```

Comment:
Hum, maybe only raise ImportError if !PyErr_Occurred()? What do you think?

I'm thinking at _PyImport_GetModuleWithError().

Reply:
The point is to keep the semantics the same.  In this case we were using PyDict_GetItem(), so I switched to _PyImport_GetModule().  I don't want to change semantics here, regardless of how small a change.